client/ayon_core/tools/sceneinventory/view.py

```diff
@@ -747,12 +754,25 @@ def _show_version_dialog(self, item_ids, active_repre_id):
             repre_info.product_id
             for repre_info in repre_info_by_id.values()
         }
+        project_products = {}
+        for container_item in container_items_by_id.values():
+            repre_id = container_item.representation_id
+            project_name = container_item.project_name
+            repre_info = repre_info_by_id.get(repre_id)
+            if repre_info and repre_info.is_valid:
+                if project_name not in project_products:
+                    project_products[project_name] = set()
+                    product_id = repre_info.product_id
+                    project_products[project_name].add(product_id)
+        print("proj_product", project_products)
         active_repre_info = repre_info_by_id[active_repre_id]
         active_version_id = active_repre_info.version_id
         active_product_id = active_repre_info.product_id
-        version_items_by_product_id = self._controller.get_version_items(
-            product_ids
-        )
+        version_items_by_product_id = {}
+        for project_name, product_ids in project_products.items():
+            version_items_by_product_id.update(
+                self._controller.get_version_items(
+                project_name, product_ids))
```

Comment:
Much easier to read...
```suggestion
            version_items = self._controller.get_version_items(
                project_name, product_ids
            )
            version_items_by_product_id.update(version_items)
                
```